vigenere-cipher/src/App.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from "react";
+
+const App: React.FC = () => {
+  const [inputText, setInputText] = useState("");
+  const [resultText, setResultText] = useState("");
+  const [encryptionKey, setEncryptionKey] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const alphabet = [
+    "a",
+    "b",
+    "c",
+    "d",
+    "e",
+    "f",
+    "g",
+    "h",
+    "i",
+    "j",
+    "k",
+    "l",
+    "m",
+    "n",
+    "o",
+    "p",
+    "q",
+    "r",
+    "s",
+    "t",
+    "u",
+    "v",
+    "w",
+    "x",
+    "y",
+    "z",
+  ];
+
+  function onChangeHandler(element: string, value: string) {
+    if (validator(value)) {
+      setErrorMessage("");
+      if (element === "inputText") {
+        setInputText(value);
+      } else {
+        setEncryptionKey(value);
+      }
+    } else {
+      setErrorMessage("小文字のアルファベットのみ入力可能です");
+    }
+  }
+
+  function validator(value: string) {
+    const regex = /[^a-z]/g;
+    const found = value.match(regex);
+    if (found && found.length >= 1) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function onClickHandler(element: string) {
```

Comment:
element は何かよくわかんないから、mode とか method とか direction とかのほうがいいかな〜
```suggestion
  function onClickHandler(mode: string) {
```